src/js/contentScript.js

```diff
@@ -107,25 +104,57 @@ const domInspector = (function ()
     return _inspectElements();
 });
 
-/*
+/**
+ * Checks whether the page is supported by the plugin
+ * @param hostname URL of the page
+ * @return {boolean} Whether page is supported or not
+ **/
+const _isUnsupportedPage = function (hostname)
+{
+    var unsupportedPagesList = [TWITTER, FACEBOOK];
+    return unsupportedPagesList.some(function (page)
+    {
+        return hostname.includes(page);
+    });
+};
+
+/**
+ * Checks if the site is supported by the plugin and performs the necessary actions -
+ * either warns user of this or if supported, carries on as usual
+ */
+const _checkSupported = function ()
+{
+    if (_isUnsupportedPage(window.location.hostname)) {
+        var warnedUnsupported = localStorage.getItem('warnedUnsupported') || '';
+        if (warnedUnsupported !== 'yes') {
+            alert(WARN_UNSUPPORTED_PAGE);
+            localStorage.setItem('warnedUnsupported','yes');
```

Comment:
A few things:
1. Can you put these strings as constants or does that break the program?
2. localStorage persists, would there be no warnings if the page was reloaded?
3. Not sure you need || '' bit since the condition check will fail anyway if getItem returns null
4. Perhaps change 'yes' to 'true'? That would be potentially reusable in the future